### ISSUE TITLE
aria-hidden="true" not obeyed?

### ISSUE THREAD

**Issue by j1mmie** [ken107/read-aloud#438]
Seems like the extension doesn't consider the `aria-hidden` attribute at all. `aria-hidden` is fairly well defined and has existed for 10+ years, so I'm a bit surprised. Maybe I'm doing something wrong?

Here's a test HTML file:

```
<html>
  <body>
    <div aria-hidden="true">This content should not be read by the screen reader</div>

    <main>
      <h1>Hello</h1>
      <p>But this is actually the real content</p>
    </main>
  </body>
</html>
```

This results in the following TTS text:

![Image](https://github.com/user-attachments/assets/d787ebc3-97a0-4dc1-bcd8-e4eb6d45ffdd)

The only way I can seem to get read-aloud to ignore visible content is to mark it as `display: none`, which I do not want to do for my visual users.

**Comment by j1mmie** [comment]
Ah, I see this was addressed in August last year, in this commit:

https://github.com/ken107/read-aloud/commit/0a00f366f01f81af457666692597baa896632730

Here I've also learned about the class `no-read-aloud`, which is non-standard, but works for me for now

**Comment by ken107** [comment]
Yes, that commit should have caused aria-hidden content to be ignored. I'll need to see why it's not working. Thanks

**Comment by j1mmie** [comment]
I think my extension was just out of date since the last release was 2022 and the fix went in in 2024

**Comment by ken107** [comment]
Thats probably it then, will close.
